src/math.F90

```diff
@@ -702,6 +718,95 @@ function watt_spectrum(a, b) result(E_out)
 
   end function watt_spectrum
 
+!===============================================================================
+! FADDEEVA the Faddeeva function, using Stephen Johnson's implementation
+!===============================================================================
+
+  function faddeeva(z) result(wv)
+    complex(C_DOUBLE_COMPLEX), intent(in) :: z ! The point to evaluate Z at
+    complex(8)     :: wv     ! The resulting w(z) value
+    real(C_DOUBLE) :: relerr ! Target relative error in inner loop of MIT
+                             !  Faddeeva
+
+    ! Technically, the value we want is given by the equation:
+    ! w(z) = I/Pi * Integrate[Exp[-t^2]/(z-t), {t, -Infinity, Infinity}]
+    ! as shown in Equation 63 from Hwang, R. N. "A rigorous pole
+    ! representation of multilevel cross sections and its practical
+    ! applications." Nuclear Science and Engineering 96.3 (1987): 192-209.
+    !
+    ! The MIT Faddeeva function evaluates w(z) = exp(-z^2)erfc(-iz). These
+    ! two forms of the Faddeeva function are related by a transformation.
+    !
+    ! If we call the integral form w_int, and the function form w_fun:
+    ! For imag(z) > 0, w_int(z) = w_fun(z)
+    ! For imag(z) < 0, w_int(z) = -conjg(w_fun(conjg(z)))
+
+    relerr = ZERO
+    if (aimag(z) > ZERO) then
+      wv = faddeeva_w(z, relerr)
+    else
+      wv = -conjg(faddeeva_w(conjg(z), relerr))
+    end if
+
+  end function faddeeva
+
+!===============================================================================
+! BROADEN_WMP_POLYNOMIALS Doppler broadens the windowed multipole curvefit.  The
+! curvefit is a polynomial of the form
+! a/En + b/sqrt(En) + c + d sqrt(En) ...
```

Comment:
It would be great to put a reference in this comment to the JCP paper so someone looking at the source can figure out where these equations come from.